include/curl/curl.h

```diff
@@ -65,7 +65,8 @@
 #include <stdio.h>
 #include <limits.h>
 
-#if (defined(__FreeBSD__) && (__FreeBSD__ >= 2)) || defined(__MidnightBSD__)
+#if (defined(__FreeBSD__) && (__FreeBSD__ >= 2) && (__FreeBSD__ < 8)) || \
+  defined(__MidnightBSD__)
```

Comment:
```suggestion
#if defined(__FreeBSD__) && (__FreeBSD__ >= 8)
#include <sys/param.h>
#elif (defined(__FreeBSD__) && (__FreeBSD__ >= 2) && (__FreeBSD__ < 8)) || \
  defined(__MidnightBSD__)
```

from the issue

> FreeBSD manual [says](https://docs.freebsd.org/en/books/porters-handbook/porting-dads/#porting-versions) use `#include <sys/param.h>`.

It may apply to Midnight BSD as well but I can't find any documentation on it.

Nobody else has reported this which is surprising. I think maybe the reporter had some bad cross compile, and maybe osreldate still exists. It seems like we would've heard about it already.

Reply:
osreldate.h still exists on modern FreeBSD versions; it hasn't been removed.